base/promotion.jl

```diff
@@ -122,14 +122,14 @@ end
 
 ## promotion mechanism ##
 
-promote_type()  = (@_pure_meta; Bottom)
-promote_type(T) = (@_pure_meta; T)
-promote_type(T, S, U, V...) = (@_pure_meta; promote_type(T, promote_type(S, U, V...)))
+promote_type()  = (@_inline_meta; Bottom)
```

Comment:
Nope - so far it's been mostly an exercise of find-replace. I just wanted to quickly get it working and check with Jameson if there was a reason *not* to do this, but I'll tidy it up soon.